builds-cleaner.groovy

```diff
@@ -0,0 +1,30 @@
+/*** BEGIN META {
+  "name" : "Builds Cleaner",
+  "comment" : "Delete all builds keeping ${MAX_BUILDS} from all job of the Jenkins Instance. It checks build is not building and it also keeps the keep forever, lastStableBuild, lastSuccessfulBuild, lastSuccessfulBuild, lastUnstableBuild, lastUnsuccessfulBuild in case are not included in the ${MAX_BUILDS}.",
+  "parameters" : [],
+  "core": "2.107.3",
+  "authors" : [
+    { name : "Marco Davalos" }
+  ]
+} END META**/
+
+import jenkins.model.Jenkins
+import hudson.model.Job
+
+def MAX_BUILDS = 5 // max builds to keep
```

Comment:
Update `def MAX_BUILDS` to `int MAX_BUILDS`.  While groovy allows use of `def`, it is still a good Groovy practice to enforce strong typing on variables which are intended to be only one type.  For example, I don't think you ever expect `MAX_BUILDS` to be a `String` or a `HashMap`.  So it should be an `int` or `Integer`; which type you use doesn't matter because Groovy translates both to `Integer` when it interprets the script.  I usually use `int` because it is fewer characters but I don't really care which you decide to use as long as it is strongly typed.